main.nf

```diff
@@ -265,143 +199,206 @@ if (params.part == 2) {
   }
 }
 
-if (params.bindir) ch_input_files = Channel.fromPath("${params.bindir}/*")
-if (params.rbindir) ch_input_files = Channel.fromPath("${params.rbindir}/*")
+if (params.step =~ 4 || params.step =~ 5 || params.step =~ 6) {
+  if (params.bindir) {
+    ch_input_files = Channel.fromPath("${params.bindir}/*")
+  } else if (params.rbindir) {
+    ch_input_files = Channel.fromPath("${params.rbindir}/*")
+  } else if (params.step =~ 2) {
+    def bin_path = ch_bin.view().val
+    ch_input_files = Channel.fromPath("${bin_path}/*")
+  } else {
+    log.error "Please provide --bindir or run step 2 first"
+    exit 1
+  }
 
-println "Total number of samples in bin directory - "
-number_of_input_files = ch_input_files.count().view().val
-number_of_batches = (int) Math.ceil(number_of_input_files/params.target_size)
-println "Number of batches to run - "
-println number_of_batches
+  println "Total number of samples in bin directory - "
+  number_of_input_files = ch_input_files.count().view().val
```

Comment:
**just a comment** to be kept in mind, this will not work in DSL2 as `.val` does not exist anymore unfortunately